docs/mgmt/mitigate_breaking_changes.md

```diff
@@ -0,0 +1,108 @@
+1. The SDK is designed to support multiple API versions, therefore we should ensure that the mitigation steps do not introduce breaking changes for older API versions.
+
+## Mitigation Steps
+
+### `MembersMustExist` Errors
+
+`MembersMustExist` errors from ApiCompat are usually seen when a member is removed in the new version of the SDK. This can happen when a method/property/class is removed, or when a method/property/class is renamed. To mitigate this, we should introduce the removed member back and add the `[EditorBrowsable(EditorBrowsableState.Never)]` attribute to the member. This will hide the member from intellisense, and will prevent the member from being used in client code. This will not be a breaking change for clients that are using the member, but it will prevent new clients from using the member.
+
+To mitigate a `MembersMustExist` error for removed **properties** or **methods**, follow these steps:
+
+1. Create a file outside the `src/Generated` directory with the same name of its containing class. Usually they should go into the `src/Customization` or `src/Custom` directory, if there is not such a directory, please create one.
+1. Create a partial class in the file with the same name as the containing class.
+1. Introduce the property or method back (**including its original implementation and xml documentation**) with the `[EditorBrowsable(EditorBrowsableState.Never)]` attribute.
+1. Introduce everything needed in the original implementation back as well, including any helper methods, classes, etc.
+
+In addition, for removed properties, we need to do some additional steps to make sure our SDK still works for old API versions. Please see the section below for more details.
+
+If the property no longer works in all API versions, please add the `[Obsolete("<Put a message here>", false)]` attribute to the property and just leave the property with its auto getter and setter.
+
+```csharp
+// in file `src/Customization/TheTypeContainingRemovedProperty.cs`
+public partial class TheTypeContainingRemovedProperty
+{
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    [Obsolete("This property no longer works in all API versions. Please use the <NewProperty> instead.", false)]
+    public string RemovedProperty { get; } // if it previously does not have a setter
+    //public string RemovedProperty { get; set; } // if it previously has a setter
+}
+```
+
+If the property still works for old API versions, and it no longer works for newer API versions, please **DO NOT** add the `Obsolete` attribute to this property. In addition, to make sure our SDK still works for old API versions, please add a new internal constructor that takes all the properties including this removed property, and call the base generated internal constructor, and we need to add the removed property to its serialization method and deserialization method (if any).
```

Comment:
We should throw NotSupportedException if an API does not work. If a property does not work only in some API versions, we should throw conditionally.

Reply:
Also, we should discuss the point about not obsoleting. I think it would be better to obsolete if the newest API version does not work. If only older versions don't work, then I agree that we should not obsolete.